cmd/argchecker.go

```diff
@@ -0,0 +1,29 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+)
+
+func verifyArguments(cmd *cobra.Command) error {
+	gwString, _ := cmd.Flags().GetString("gateway")
+	dbString, _ := cmd.Flags().GetString("dashboard")
+
+	if gwString == "" && dbString == "" {
+		return errors.New(fmt.Sprintf("%s requires either gateway or dashboard target to be set", cmd.Use))
+	}
+
+	if gwString != "" && dbString != "" {
+		return errors.New(fmt.Sprintf("%s requires either gateway or dashboard target to be set, not both", cmd.Use))
+	}
+
+	brString, _ := cmd.Flags().GetString("branch")
+	ptString, _ := cmd.Flags().GetString("path")
+
+	if ptString != "" && brString != "refs/heads/master" {
+		return errors.New(fmt.Sprintf("%s requires either files or branch to be set, not both", cmd.Use))
+	}
```

Comment:
Isn't easier to just define a hierarchy and stick to that instead of all this logic to check that only one parameter is set?
Something like 
```go
         if gwString != "" {
                 fmt.Sprintf("gwString parameter found")

                 dbString == ""
		
	} else {
                    if dbString == "" {
                     return errors.New(fmt.Sprintf("%s requires either gateway or dashboard target to be set", cmd.Use))   
                    }
        }
```

Reply:
No, using guard clauses is preferred over nested conditionals

https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html